api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -1059,6 +1103,65 @@ public void saveRole_shouldThrowErrorIfRoleInheritsFromItself() {
 		userService.saveRole(parentRole);
 	}
 	
+	@Test
+	public void saveRole_shouldAllowARoleToBeSavedWithCorrectPermissions() throws IllegalAccessException {
+		Role role = new Role("my role");
+		Privilege myPrivilege = new Privilege("custom privilege");
+		role.addPrivilege(myPrivilege);
+		
+		User currentUser = new User();
+		currentUser.addRole(new Role(RoleConstants.SUPERUSER));
+		
+		withCurrentUserAs(currentUser, () -> {
+			Role newRole = new Role("another role");
+			newRole.addPrivilege(myPrivilege);
+			userService.saveRole(newRole);
+		});
+	}
+
+	@Test
+	public void saveRole_shouldThrowErrorWhenCurrentUserLacksPrivilegeAssignedToRole() throws IllegalAccessException {
+		Role adminRole = new Role("my role");
+		adminRole.addPrivilege(new Privilege(PrivilegeConstants.MANAGE_ROLES));
+
+		User currentUser = new User();
+		currentUser.addRole(adminRole);
+
+		Privilege myPrivilege = new Privilege("custom privilege");
+		
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("You must have the following privileges in order to assign them: custom privilege");
+
+		withCurrentUserAs(currentUser, () -> {
+			Role newRole = new Role("another role");
+			newRole.addPrivilege(myPrivilege);
+			userService.saveRole(newRole);
+		});
+	}
+
+	@Test
+	public void saveRole_shouldThrowErrorWhenCurrentUserLacksAPrivilegeAssignedToRole() throws IllegalAccessException {
+		Privilege myFirstPrivilege = new Privilege("custom privilege");
+		Privilege mySecondPrivilege = new Privilege("another privilege");
+		
+		Role adminRole = new Role("my role");
+		adminRole.addPrivilege(new Privilege(PrivilegeConstants.MANAGE_ROLES));
+		adminRole.addPrivilege(myFirstPrivilege);
+
+		User currentUser = new User();
+		currentUser.addRole(adminRole);
+
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage("You must have the following privileges in order to assign them: another privilege");
+
+		withCurrentUserAs(currentUser, () -> {
+			Role newRole = new Role("another role");
+			newRole.addPrivilege(myFirstPrivilege);
+			newRole.addPrivilege(mySecondPrivilege);
+			userService.saveRole(newRole);
+		});
+	}
+	
```

Comment:
Why is this in this PR?

Reply:
This again came with the same PR but before this test was earlier better, so i guess it has no conflit